developer-docs-site/docs/standards/fungible-asset.md

```diff
@@ -0,0 +1,403 @@
+
+Minimally, Aptos coin should be interchangeable with FA, it is up to the will of the ecosystem to dictate the whether or not Aptos coin is replaced by a FA equivalent.
+
+## Structures
+
+### Metadata Object
+
+Within FA, metadata defines attributes of the type and other common features. The type, itself, is defined by the Object or address where this information lies. In other words, two assets with identical metadata but distinct Objects are not the same. The metadata layout is defined as:
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+struct Metadata has key {
+    supply: Option<Supply>,
+    /// Name of the fungible metadata, i.e., "USDT".
+    name: String,
+    /// Symbol of the fungible metadata, usually a shorter version of the name.
+    /// For example, Singapore Dollar is SGD.
+    symbol: String,
+    /// Number of decimals used for display purposes.
+    /// For example, if `decimals` equals `2`, a balance of `505` coins should
+    /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+    decimals: u8,
+}
+```
+
+### Fungible Asset and Fungible Store
+
+FA allows typing to be decoupled from metadata by allocating an object reference that points at the metadata. Hence a set of units of FA is represented as an amount and a reference to the metadata, as shown:
+
+```rust
+struct FungibleAsset {
+    metadata: Object<Metadata>,
+    amount: u64,
+}
+```
+
+In contrast to Objects and addresses, a Coin uses a generic, or the `CoinType`, to support distinct typing within the Coin framework. For example, `Coin<A>` and `Coin<B>` are two distinct coins, if `A != B`.
+
+The fungible assets is a struct representing the type and the amount of units held. As the struct does not have either key or store abilities, it can only be passed from one function to another but must be consumed by the end of a transaction. Specifically, it must be deposited back into a fungible store at the end of the transaction:
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct FungibleStore has key {
+    /// The address of the base metadata object.
+    metadata: Object<Metadata>,
+    /// The balance of the fungible metadata.
+    balance: u64,
+    /// Fungible Assets transferring is a common operation, this allows for freezing/unfreezing accounts.
+    frozen: bool,
+}
+```
+
+The only extra field added here is `frozen`. if it is `true`, this object is frozen, i.e. deposit and withdraw are both disabled without using `TransferRef` in the next section.
+
+### References
+
+Reference(ref) is the means to implement permission control across different standard in Aptos. In different contexts, it may be called capabilities sometimes. In FA standard, there are three distinct refs one each for minting, transferring, and burning FA respectively called `MintRef`, `TransferRef`, and `BurnRef`. Each ref contains a reference to the FA metadata:
```

Comment:
```suggestion
_Reference_ (ref) is the means to implement permission control across different standards in Aptos. In different contexts, it may be called _capabilities_. The FA standard has three distinct refs for minting, transferring, and burning FA: `MintRef`, `TransferRef`, and `BurnRef`. Each ref contains a reference to the FA metadata:
```